packages/core/src/App/Containers/Modals/app-modals.jsx

```diff
@@ -44,11 +47,28 @@ const AppModals = ({
     is_close_uk_account_modal_visible,
     is_eu,
     is_logged_in,
+    should_show_cooldown_warning_modal,
+    should_show_assessment_complete_modal,
+    is_trading_assessment_for_new_user_enabled,
+    fetchFinancialAssessment,
+    setCFDScore,
+    cfd_score,
+    active_account_landing_company,
     is_deriv_account_needed_modal_visible,
 }) => {
     const url_params = new URLSearchParams(useLocation().search);
     const url_action_param = url_params.get('action');
 
+    React.useEffect(() => {
+        const fetchFinancialScore = async () => {
```

Comment:
Do we need to declare this function if we only use it once?

Reply:
Will need to check  how an async function can be called inside useEffect

Reply:
if (is_logged_in) {
        fetchFinancialAssessment().then(response => setCFDScore(response?.cfd_score ?? 0));
}
* this will eliminate the need to declare the function😊